.github/workflows/codeql-analysis.yml

```diff
@@ -37,18 +37,23 @@ jobs:
         # https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#changing-the-languages-that-are-analyzed
 
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v2
-
-    # Initializes the CodeQL tools for scanning.
-    - name: Initialize CodeQL
-      uses: github/codeql-action/init@v1
-      with:
-        languages: ${{ matrix.language }}
-        # If you wish to specify custom queries, you can do so here or in a config file.
-        # By default, queries listed here will override any specified in a config file.
-        # Prefix the list here with "+" to use these queries and those in the config file.
-        # queries: ./path/to/local/query, your-org/your-repo/queries@main
-
-    - name: Perform CodeQL Analysis
-      uses: github/codeql-action/analyze@v1
+      - name: Harden Runner
+        uses: step-security/harden-runner@14dc64f30986eaa2ad2dddcec073f5aab18e5a24 # v1
+        with:
+          egress-policy: audit
```

Comment:
As discussed internally, could we get rid of this external dependency? I acknowledge we'd be less hardened in terms of the calls the action does, but we wouldn't be introducing an additional external dependency.

Reply:
As the result of the action already show, we already have several dependencies, and actions that reach out to third-party sites - one more (that gives us insight into those we already use) isn't going to be detrimental.

I don't feel strongly though, so if you do feel stronger, it can be removed.

Reply:
Sorry for the late response, this completely slipped through my radar. 

I would still remove this external dependency. Other than that the PR looks great.

Reply:
Hi @pauarge, can you please help me understand the concern? You don't have to use it, but understanding the concern might help me improve it. In an end-to-end test workflow, adding `harden-runner` is not going to compromise the build output or anything. Best case you might find a compromised dependency making outbound calls (which they typically do), but what is the worst case of using `harden-runner` in an end-to-end test workflow?

Reply:
Hi @varunsh-coder, thanks for reaching out. I am not particularly keen on a third-party dependency running on our CI that's actively keeping a log of our builds: https://app.stepsecurity.io/github/Parsely/wp-parsely/actions/runs/1704105637

Reply:
> Hi @varunsh-coder, thanks for reaching out. I am not particularly keen on a third-party dependency running on our CI that's actively keeping a log of our builds: https://app.stepsecurity.io/github/Parsely/wp-parsely/actions/runs/1704105637

Thanks a lot @pauarge for the feedback! Would it help if: 
1. The logs were not publicly visible, but were private only to those who have access to the repo? This is in `preview` for [private repositories](https://github.com/step-security/harden-runner#support-for-private-repositories), but I could also enable it for Public repositories. AND/OR
2. The logs expired and were purged in a few days, e.g. 1 day. AND/OR
3. In `block` mode, you could configure it to not send out telemetry. This was a request from another developer using `harden-runner`, and is being tracked [here](https://github.com/step-security/harden-runner/issues/89). 

Please let me know which of these options will help with your concern or if you have other ideas. The goal is really to help developers find compromised dependencies/ build tools, and if anything in the method is causing concern with that goal, I can try to address that. Thanks again for the feedback!